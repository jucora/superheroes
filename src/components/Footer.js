import React from 'react';

function Footer() {
  return (
    <div className="footerContainer">
      <footer>
        <p>Julian Belmonte copyright 2020</p>
      </footer>
    </div>
  );
}

export default Footer;
