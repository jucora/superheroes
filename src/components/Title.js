import React from 'react';
import { NavLink, HashRouter as Router } from 'react-router-dom';

const Title = () => (
  <Router>
    <div className="title row">
      <header>
        <h1 >SUPERHEROES</h1>
        <div  className="title-subtitle">
          Search and get information about your favorite superhero!
        </div>
        <div>
          <NavLink
            className="nav-link"
            activeClassName="nav-link-active"
            exact
            to="/"
          >
            List
          </NavLink>
          <NavLink
            className="nav-link"
            activeClassName="nav-link-active"
            to="/about"
          >
            About
          </NavLink>
        </div>
      </header>
    </div>
  </Router>
);

export default Title;
