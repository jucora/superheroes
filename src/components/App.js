import React from "react";
import { HashRouter as Router, Route } from "react-router-dom";
import { Provider } from "react-redux";
import { createStore } from "redux";
import Title from "./Title";
import HeroList from "../containers/HeroList";
import Footer from "./Footer";
import Author from "./Author";
import HeroDetails from "../containers/HeroDetails";
import HeroForm from "../containers/HeroForm";
import storage from "../localStorage/storage";
import rootReducer from "../reducers/index";

const persistState = storage.loadState();

let store = createStore(rootReducer, persistState());

store.subscribe(() => {
  storage.saveState(store.getState());
});

const App = () => (
  <Provider store={store}>
    <Router>
      <div className="App">
        <Title />
        {window.location.pathname === "/" ? <HeroForm /> : null}
        <Route exact path="/" component={HeroList} />
        <Route path="/about" component={Author} />
        <Route path="/details/:id" component={HeroDetails} />

        <Footer />
      </div>
    </Router>
  </Provider>
);

export { store };
export default App;
