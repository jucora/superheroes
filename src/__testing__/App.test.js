import { HashRouter as Router, Route } from 'react-router-dom';
import App from '../components/App';
import Title from '../components/Title';
import Footer from '../components/Footer';


it('renders correctly', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});

it('should contain a Router', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Router).length).toBe(1);
});

it('renders Title component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Title).length).toBe(1);
});

it('renders 3 Route components', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Route).length).toBe(3);
});
it('renders Footer component', () => {
  const wrapper = shallow(<App />);
  expect(wrapper.find(Footer).length).toBe(1);
});
