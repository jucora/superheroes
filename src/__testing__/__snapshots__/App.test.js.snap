// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<HashRouter>
  <Title />
  <Switch>
    <Route
      component={[Function]}
      path="/about"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Details)",
          "type": [Function],
        }
      }
      path="/details/:id"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Home)",
          "type": [Function],
        }
      }
      exact={true}
      path="/"
    />
  </Switch>
  <Footer />
</HashRouter>
`;
